pkg/db/postgres.go

```diff
@@ -0,0 +1,145 @@
+package db
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+// Pool is the global, exported database connection pool instance.
+// It will be nil if the connection failed.
+var Pool *pgxpool.Pool
+
+// InitPool attempts to initialize the PostgreSQL connection pool.
+// It logs a warning if initialization fails and sets Pool to nil, allowing the app to continue.
+func InitPool() {
+	connStr := os.Getenv("DATABASE_URL")
+	if connStr == "" {
+		// Log warning and return if the environment variable is not set
+		log.Println("WARNING: DATABASE_URL environment variable is not set. Database functions will be disabled.")
+		Pool = nil
+		return
+	}
+
+	config, err := pgxpool.ParseConfig(connStr)
+	if err != nil {
+		log.Printf("WARNING: Unable to parse DATABASE_URL configuration: %v. Database functions will be disabled.", err)
+		Pool = nil
+		return
+	}
+
+	// Configure connection pool settings
+	config.MaxConns = 25
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Attempt to create the connection pool
+	pool, err := pgxpool.NewWithConfig(ctx, config)
+	if err != nil {
+		log.Printf("WARNING: Unable to create connection pool: %v. Database functions will be disabled.", err)
+		Pool = nil
+		return
+	}
+
+	// Attempt to ping the database
+	if err = pool.Ping(ctx); err != nil {
+		log.Printf("WARNING: Database connection failed to ping: %v. Database functions will be disabled.", err)
+		// Close the temporary pool if ping failed, before setting the global Pool to nil
+		pool.Close()
+		Pool = nil
+		return
+	}
+
+	// Success
+	Pool = pool
+	log.Println("PostgreSQL connection pool successfully initialized.")
+
+	// Initialize database schema
+	if err := InitSchema(ctx); err != nil {
+		log.Printf("WARNING: Failed to initialize database schema: %v. Some features may not work.", err)
+	}
```

Comment:
Should we fail hard in this case? If we break something changing the schema, we might not realise and stop tracking metrics, if we fail hard and exit we wouldn't be able to start the pod at all and breaking changes would be less likely to make it to prod